tests/codegen_testcases/solidity/borsh_decoding_complex_types.sol

```diff
@@ -139,99 +127,95 @@ contract Testing {
     function complexArray(bytes memory buffer) public {
         NonConstantStruct[] memory arr = abi.decode(buffer, (NonConstantStruct[]));
 
-	    // CHECK: ty:uint32 %temp.32 = (builtin ArrayLength ((arg #0)))
-	    // CHECK: ty:uint32 %temp.34 = uint32 0
-	    // CHECK: branchcond (uint32 4 <= %temp.32), block1, block2
-    
-        // CHECK: block1: # inbounds
-    	// CHECK: ty:uint32 %temp.35 = (builtin ReadFromBuffer ((arg #0), uint32 0))
-	    // CHECK: ty:uint32 %temp.34 = uint32 4
-	    // CHECK: ty:struct Testing.NonConstantStruct[] %temp.36 = (alloc struct Testing.NonConstantStruct[] len %temp.35)
-	    // CHECK: ty:struct Testing.NonConstantStruct[] %temp.33 = %temp.36
-	    // CHECK: ty:uint32 %for_i_0.temp.37 = uint32 0
-	    // CHECK: branch block3
-        
-        // CHECK: block2: # out_of_bounds
-	    // CHECK: assert-failure
-        
-        // CHECK: block3: # cond
-	    // CHECK: branchcond (unsigned less %for_i_0.temp.37 < (builtin ArrayLength (%temp.33))), block5, block6
-        
-        // CHECK: block4: # next
-	    // CHECK: ty:uint32 %for_i_0.temp.37 = (%for_i_0.temp.37 + uint32 1)
-	    // CHECK: branch block3
+		// CHECK: block0: # entry
```

Comment:
This is the validation thing again. Please, do not just copy and paste new codegen code assuming it is all correct. This test is here to ensure the correct flow path exist when we build the contract.